# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/swgl/src/glsl.h
# Commit: c0f58fd74e56
# Full Hash: c0f58fd74e56b24f67fccae08097d151e21a0f80
# Author: Ashley Hale <ahale@mozilla.com>
# Date: 2024-05-09 16:15:25
# Regressor Bug: 1824502
# File Overlap Count: 1
# Description:
#   Bug 1824502 - Implement native SVG filter rendering in WebRender r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D209646
# ==============================================================================

diff -r 3e2ec3a3b738 -r c0f58fd74e56 gfx/wr/swgl/src/glsl.h
--- a/gfx/wr/swgl/src/glsl.h	Thu May 09 02:12:39 2024 +0000
+++ b/gfx/wr/swgl/src/glsl.h	Thu May 09 04:09:00 2024 +0000
@@ -106,6 +106,7 @@
 
 int32_t force_scalar(I32 i) { return i[0]; }
 
+struct vec3;
 struct vec4;
 struct ivec2;
 
@@ -2287,6 +2288,12 @@
   return bvec2_scalar(notEqual(x.x, y.x), notEqual(x.y, y.y));
 }
 
+vec3 floor(vec3 v) { return vec3(floor(v.x), floor(v.y), floor(v.z)); }
+
+vec4 floor(vec4 v) {
+  return vec4(floor(v.x), floor(v.y), floor(v.z), floor(v.w));
+}
+
 struct mat4_scalar;
 
 struct mat2_scalar {